packages/extension-base/src/background/handlers/Tabs.ts

```diff
@@ -19,16 +20,27 @@ import RequestExtrinsicSign from '../RequestExtrinsicSign';
 import State from './State';
 import { createSubscription, unsubscribe } from './subscriptions';
 
-function transformAccounts (accounts: SubjectInfo): InjectedAccount[] {
+function getSortedAccounts (accounts: SubjectInfo): SingleAddress[] {
   return Object
     .values(accounts)
     .filter(({ json: { meta: { isHidden } } }) => !isHidden)
-    .sort((a, b) => (a.json.meta.whenCreated || 0) - (b.json.meta.whenCreated || 0))
+    .sort((a, b) => (a.json.meta.whenCreated || 0) - (b.json.meta.whenCreated || 0));
+}
+
+function transformAccounts (accounts: SubjectInfo): InjectedAccount[] {
+  return getSortedAccounts(accounts)
     .map(({ json: { address, meta: { genesisHash, name } } }): InjectedAccount => ({
       address, genesisHash, name
     }));
 }
 
+function transformAllAccounts (accounts: SubjectInfo): [KeypairType, InjectedAccount][] {
+  return getSortedAccounts(accounts)
+    .map(({ json: { address, meta: { genesisHash, name } }, type }) => ([
+      type as KeypairType, { address, genesisHash, name }
```

Comment:
type can be undefined here, hence the casting, will this bite any time or do you think `SubjectInfo` will always have it?